modAionImpl/src/org/aion/zero/impl/pendingState/AionPendingStateImpl.java

```diff
@@ -458,85 +474,73 @@ private void fireTxUpdate(AionTxReceipt txReceipt, PendingTransactionState state
      * @return SUCCESS if transaction gets NEW_PENDING state, else appropriate message such as
      *     DROPPED, INVALID_TX, etc.
      */
-    private TxResponse addPendingTransactionImpl(final AionTransaction tx) {
+    private TxResponse addPendingTransactionInner(final AionTransaction tx) {
 
-        AionTxExecSummary txSum;
-        boolean ip = inPool(tx.getNonceBI(), tx.getSenderAddress());
-        if (ip) {
+        if (txPool.isContained(tx.getSenderAddress(), tx.getNonceBI())) {
             // check energy usage
             PooledTransaction poolTx = txPool.getPoolTx(tx.getSenderAddress(), tx.getNonceBI());
             if (poolTx == null) {
                 LOGGER_TX.error(
-                        "addPendingTransactionImpl no same tx nonce in the pool {}", tx.toString());
-                fireDroppedTx(tx, "REPAYTX_POOL_EXCEPTION");
-                return TxResponse.REPAYTX_POOL_EXCEPTION;
+                    "The pool data has broken, missing the transaction! sender: {}, nonce: {}",
+                    tx.getSenderAddress(),
+                    tx.getNonceBI());
+                throw new IllegalStateException();
```

Comment:
This exception should have some text explaining the issue instead of relying on the log. Also, is this check necessary? Wouldn't the `isContained` already guarantee that the transaction exists? If not, the `if` could be removed instead. And the code can be simplified to check `poolTx != null`.